crates/shared/src/order_validation.rs

```diff
@@ -1059,6 +1066,22 @@ mod tests {
             .validate_and_construct_order(creation.clone(), &domain_separator, Default::default())
             .await
             .is_ok());
+
+        let creation = OrderCreation {
+            data: OrderData {
+                fee_amount: U256::zero(),
+                ..creation.data
+            },
+            ..creation
+        };
+        let limit_order_enabled_validator = validator.with_limit_orders(true);
+        let result = limit_order_enabled_validator
+            .validate_and_construct_order(creation.clone(), &domain_separator, Default::default())
+            .await;
+        assert!(result.is_ok());
```

Comment:
This assertion is redundant because `result.unwrap()` already causes the test to fail if `result.is_err()`.

Reply:
I'm aware but I was trying to be more explicit. Will remove this.